elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -1214,6 +1214,36 @@ public boolean hasBinding(Class<?> cls) {
         return bindJsonApiToEntity.contains(cls);
     }
 
+    /**
```

Comment:
A few general comments (I couldn't find one specific place to put these):
1.  We should rely on the EntityDictionary less throughout.  We should build the schema and then use the schema in place of the dictionary where ever possible.
2. Reflection (checking for presence of annotations) is called repeatedly.  Reflection is pretty costly.  We should use the schema instead and its related classes (metric & dimension) to answer questions rather than going to the dictionary again and again.
3. Metric expansion has to happen at query time.  We cannot know in advance what agg function to apply during the expansion. 
4. Utils classes (IMO) are usually just misplaced functions.  I have some comments on this below, but let's find a proper home for those functions and retire the Utils class.

Reply:
I'll refactor according to these